src/client/BuilderClient.ts

```diff
@@ -318,4 +320,79 @@ export class BuilderClient {
 
     return body.data
   }
+
+  public async uploadLandRedirectionFile(
+    { x, y }: LandCoords,
+    locale: string
+  ): Promise<LandHashes> {
+    let res: Response
+
+    try {
+      res = await this.fetch(`/v1/lands/${x},${y}/redirection`, {
+        method: 'post',
+        headers: {
+          'accept-language': locale
+        }
+      })
+    } catch (e) {
+      throw new ClientError(e.message, undefined, null)
+    }
+
+    const body: ServerResponse<LandHashes> = await res.json()
+
+    if (!res.ok || !body.ok) {
+      throw new ClientError(body.error || 'Unknown error', res.status, null)
+    }
+
+    return body.data
+  }
+
+  public async getLandRedirectionHashes(
+    coordsList: LandCoords[],
+    locale: string
+  ): Promise<(LandCoords & LandHashes)[]> {
+    const output: (LandCoords & LandHashes)[] = []
+
+    const batchedCoordsList: LandCoords[][] = [[]]
+    const batchLength = 50
+
+    for (let i = 0; i < coordsList.length; i++) {
+      if ((i + 1) % batchLength === 0) {
+        batchedCoordsList.push([])
+      }
+
+      const coords = coordsList[i]
+
+      batchedCoordsList[batchedCoordsList.length - 1].push(coords)
+    }
+
+    for (const coordsList of batchedCoordsList) {
+      let res: Response
+
+      try {
+        res = await this.fetch(
+          `/v1/lands/redirection/hashes?coords=${coordsList
+            .map(({ x, y }) => `${x},${y}`)
+            .join(';')}`,
+          {
+            headers: {
+              'accept-language': locale
+            }
+          }
+        )
+      } catch (e) {
+        throw new ClientError(e.message, undefined, null)
+      }
+
+      const body: ServerResponse<(LandCoords & LandHashes)[]> = await res.json()
+
+      if (!res.ok || !body.ok) {
+        throw new ClientError(body.error || 'Unknown error', res.status, null)
+      }
+
+      body.data.forEach((d) => output.push(d))
```

Comment:
You can use `concat` for a more efficient concatenation procedure.

Reply:
Using concat